pkg/cluster/wait.go

```diff
@@ -59,6 +60,15 @@ func WaitFor(ctx context.Context, log logr.Logger, client kubernetes.Reader, clu
 			return fmt.Errorf("cluster generation (%d) and observedGeneration (%d) differ", generation, observedGeneration)
 		}
 
-		return matcher(c)
+		if err := matcher(c); err != nil {
+			count = 0
+			return err
+		}
+
+		if count < total {
+			log.V(9).Info("Cluster has reached to expected condition in %d/%d times", count, total)
+		}
```

Comment:
This part is wrong, right?
You are never increasing the count
And also you are never return an error if the count happens to be smaller than total

Reply:
We are missing a test here, you would have seen that this doesn't work if you wrote one :)

Reply:
Updated the code here. Thanks for pointing that out. I have added unit test too.